### PR TITLE
Department Suggestions

### PR DESCRIPTION
**Pull request by ImArjunAsok**

### CODE REVIEW

File: Client/src/app/patient/home/home.component.ts [hunk 1]
```diff
@@ -11,20 +11,28 @@ import { PatientService } from 'src/app/services/patient.service';
...
     private patientService: PatientService) {
 
   }
 
   ngOnInit() {
-    this.departments = this.departmentsService.getAll();
+    this.departmentsService.getAll().subscribe({
+      next: (Data) => {
+        this.departments = Data;
+        console.log(this.departments);
```
Comment (mishelshaji) [inline Client/src/app/patient/home/home.component.ts:30]: Remove console.log statements.

File: Client/src/app/patient/home/home.component.ts [hunk 2]
```diff
@@ -11,20 +11,28 @@ import { PatientService } from 'src/app/services/patient.service';
...
+      },
+      error: (err) => {
+        console.error(err);
```
Comment (mishelshaji) [inline Client/src/app/patient/home/home.component.ts:33]: Remove console.log statements.

File: Client/src/app/register/register.component.ts [hunk 3]
```diff
@@ -20,12 +20,19 @@ export class RegisterComponent {
...
   }
 
   ngOnInit() {
-    this.departments = this.DepartmentService.getAll();
+    this.departmentService.getAll().subscribe({
+      next: (Data) => {
+        this.departments = Data;
+      },
+      error: (err) => {
+        console.error(err);
```
Comment (mishelshaji) [inline Client/src/app/register/register.component.ts:33]: Remove console.log statements.

File: server/MyDoctor/MyDoctor.Service/Services/PatientService.cs [hunk 4]
```diff
@@ -9,10 +10,15 @@ namespace MyDoctor.Service.Services
...
     public class PatientService
     {
-        public async Task<ServiceResponse<bool>> PatientHomeSearch(string searchValue)
+        private readonly ApplicationDbContext _db;
+
+        public PatientService(ApplicationDbContext db)
+        {
+            _db = db;
+        }
+        public async Task<ServiceResponse<bool>> PatientHomeSearchAsync(string searchValue)
```
Comment (mishelshaji) [inline server/MyDoctor/MyDoctor.Service/Services/PatientService.cs:19]: Add a blank line above.

**Review by mishelshaji — CHANGES_REQUESTED** [review]

**Review by fathimasulthanaas — APPROVED** [review]

**Review by mishelshaji — APPROVED** [review]

**Pull request merged**